flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinBuildFrontendTask.kt

```diff
@@ -77,11 +77,30 @@ public open class VaadinBuildFrontendTask : DefaultTask() {
         if (adapter.generateBundle() && BundleValidationUtil.needsBundleBuild
                 (adapter.servletResourceOutputDirectory())) {
             BuildFrontendUtil.runFrontendBuild(adapter)
-            cleanTask.execute()
+            if (cleanFrontendFiles()) {
+                cleanTask.execute()
+            }
         }
         LicenseChecker.setStrictOffline(true)
         BuildFrontendUtil.validateLicenses(adapter)
 
         BuildFrontendUtil.updateBuildFile(adapter)
     }
+
+
+    /**
+     * Define if frontend files generated by bundle build should be cleaned or
+     * not.
+     * <p>
+     * The targeted frontend files are files that do not exist when
+     * build-frontend target is executed.
+     * <p>
+     * Extending mojo can override this method to return false so that any
+     * frontend files created for the bundle build are not removed.
+     *
+     * @return {@code true} to remove created files, {@code false} to keep files
+     */
+    protected fun cleanFrontendFiles(): Boolean {
```

Comment:
Must be declared `open` to allow sub classes to override it
```suggestion
    protected open fun cleanFrontendFiles(): Boolean {
```